scripts/core/editor3/components/embeds/EmbedBlock.tsx

```diff
@@ -5,6 +5,10 @@ import {connect} from 'react-redux';
 import {QumuWidget, isQumuWidget} from './QumuWidget';
 import * as actions from '../../actions';
 import ng from 'core/services/ng';
+import {loadIframelyEmbedJs} from './loadIframely';
+import {debounce} from 'lodash';
+
+const loadIframely = debounce(loadIframelyEmbedJs, 100);
```

Comment:
Should it be loaded more than once? If not, you shouldn't use debouncing. Make it `loaded = true` after the function call ignoring that the request is async and let it fail with the request. If we want to implement retrying, we should implement it properly. In this case, I think not implementing all it is fine.

Reply:
it should be called more than once, every time a new embed block is rendered. but on initial load if you have embeds already it would get called for each, so this just avoids that and calls it once

Reply:
`loadIframelyEmbedJs` adds the script tag to the document. Adding the script tag to the document should only be done once. You should use `lodash.once` instead of `lodash.debounce`. It may as well be done in `core/editor3/components/embeds/loadIframely.tsx` since the script is not removed through the lifetime of the application.

Reply:
I just noticed that you are trying to emulate `once` using `loaded` variable in `loadIframely.tsx`. So yeah, you can move it up and to the top of the function like this
```
let loaded = false;

export function loadIframelyEmbedJs() {
    if (loaded == true) {
        return;
    }

    loaded === true;

    const iframely = window['iframely'] = window['iframely'] || {};
    const widgets = iframely.widgets = iframely.widgets || {};
    // ...
```

and not use debounce/once or remove the variable and use `once`. There is no point of doing both.

Reply:
yep it only adds the script once, but then it has to call that `widgets.load()` every time it renders a new block